### PR TITLE
gtk2: allow configuration of gtkrc file

### LINKED ISSUE #519: Customizable gtkrc-2.0 file location
`.gtkrc-2.0` file location can be changed using `GTK2_RC_FILES` env variable. e.g.:
`export GTK2_RC_FILES="$XDG_CONFIG_HOME"/gtk-2.0/gtkrc`

It'd be very useful for people like me who annoyed by `$HOME` pollution.

### PR DESCRIPTION
**Pull request by sumnerevans**
### Description

Closes #519

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/misc/gtk.nix [hunk 1]
```diff
@@ -92,6 +92,18 @@ in {
             <filename>~/.gtkrc-2.0</filename>.
           '';
         };
+
+        configLocation = mkOption {
+          type = types.path;
+          apply = p:
+            if hasPrefix "/" p then p else "${config.home.homeDirectory}/${p}";
```
Comment (rycee) [inline modules/misc/gtk.nix:99]: I think this is unnecessary since `types.path` already implies that the path is absolute. See https://github.com/NixOS/nixpkgs/blob/dc68eb58bb8d4dc3cfb8896932e6146c45579932/lib/types.nix#L343-L347

If you want to keep the possibility of specifying a relative path, which seems reasonable, then the type will have to be `types.str`.

File: format [hunk 2]
```diff
@@ -40,7 +40,6 @@ find . -name '*.nix' \
   ! -path ./modules/services/sxhkd.nix \
   ! -path ./modules/systemd.nix \
   ! -path ./nix-darwin/default.nix \
-  ! -path ./tests/default.nix \
```
Comment (rycee) [inline format:43]: Would it be possible to move this formatting change to its own PR? If it's a hassle then no worries about keeping it here.

File: modules/misc/gtk.nix [hunk 3]
```diff
@@ -154,10 +166,12 @@ in {
     home.packages = optionalPackage cfg.font ++ optionalPackage cfg.theme
       ++ optionalPackage cfg.iconTheme;
 
-    home.file.".gtkrc-2.0".text =
+    home.file.${cfg2.configLocation}.text =
       concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
       + cfg2.extraConfig;
 
+    home.sessionVariables = { GTK2_RC_FILES = cfg2.configLocation; };
```
Comment (rycee) [inline modules/misc/gtk.nix:173]: Might as well do it without the explicit attribute set:

```suggestion
    home.sessionVariables.GTK2_RC_FILES = cfg2.configLocation;
```

**Comment by rycee** [comment]
Thanks! I've added a few comments.

File: modules/misc/gtk.nix [hunk 4]
```diff
@@ -92,6 +92,17 @@ in {
             <filename>~/.gtkrc-2.0</filename>.
           '';
         };
+
+        configLocation = mkOption {
+          type = types.path;
+          default = "${home.homeDirectory}/.gtkrc-2.0";
```
[the remark this answers is not in the document]
Comment (sumnerevans) [reply modules/misc/gtk.nix]: 🤦

**Comment by rycee** [comment]
Lovely, merged now! 👍
